ext/php8/coms.c

```diff
@@ -720,15 +721,30 @@ char *ddtrace_agent_url(void) {
         return formatted_url;
     }
 
-    return zend_strndup(ZEND_STRL("http://localhost:8126"));
+    if (access(DEFAULT_UDS_PATH, F_OK) == SUCCESS) {
+        return zend_strndup(ZEND_STRL("unix://" DEFAULT_UDS_PATH));
+    }
+
+    int64_t port = get_global_DD_TRACE_AGENT_PORT();
+    if (port <= 0 || port > 65535) {
+        port = 8126;
+    }
+    char *formatted_url;
+    asprintf(&formatted_url, HOST_FORMAT_STR, "localhost", (uint32_t)port);
+    return formatted_url;
 }
 
 void ddtrace_curl_set_hostname(CURL *curl) {
     char *url = ddtrace_agent_url();
     if (url && url[0]) {
-        size_t agent_url_len = strlen(url) + sizeof(TRACE_PATH_STR);
+        char *http_url = url;
+        if (strlen(url) > 7 && strncmp(url, "unix://", 7) == 0) {
+            curl_easy_setopt(curl, CURLOPT_UNIX_SOCKET_PATH, url + 7);
+            http_url = "http://localhost";
```

Comment:
For curl, when using UDS, does it matter what the host is? Could `localhost` be replaced with theoretically anything?